python/torch_mlir/dialects/torch/importer/jit_ir/csrc/import_options.h

```diff
@@ -23,6 +23,10 @@ struct ImportOptions {
   // a requirement to use a value-semantic tensor type (!torch.vtensor) in
   // signatures.
   bool assumeTensorsHaveValueSemantics = false;
+  // If this is set to true, then all tensors with zero rank are assumed to be
```

Comment:
Tensors with 0 rank should already be imported as [] -- if not, then that's just a bug.

Reply:
According to [this](
https://github.com/llvm/torch-mlir/blob/torch_mlir_ltc_backend/python/torch_mlir/dialects/torch/importer/jit_ir/csrc/torch_to_mlir_utils.cpp#L143-L147), if `sizes.rank()` is zero, then `optionalSizes=nullptr`. So then over [here](https://github.com/llvm/torch-mlir/blob/main/lib/CAPI/TorchTypes.cpp#L222-L223),  `optionalSizesArrayRef` remains `None`, which I found defaults to `*` instead of `[]`.

If 0 rank is _supposed_ to be `[]`, then the change would just be to default that to an empty array instead?

Reply:
but `sizes.rank()` returns an Optional
```
  c10::optional<size_t> rank() const {
    if(!dims_) {
      return c10::nullopt;
    }
    return dims_->size();
  }
```
The case of "known rank of 0" should not take the branch commented as "unranked".

It might be clearer to rewrite that code as

```
auto &symbolicShape = tensorType->symbolic_sizes();
auto &maybeSizes = symbolicShape.sizes();
if (!maybeSizes.has_value()) { unranked }
auto &sizes = maybeSizes.value();
...
```

Reply:
Yeah you're right it shouldn't go down that path, but if `sizes.rank()` is zero, then `dims.data()` will be falsy [here](https://github.com/llvm/torch-mlir/blob/torch_mlir_ltc_backend/python/torch_mlir/dialects/torch/importer/jit_ir/csrc/torch_to_mlir_utils.cpp#L158), which will result in a shape of `*` instead of `[]`

Reply:
Yeah, I think I have run into this before -- std::vector doesn't guarantee that dims.data() is a valid pointer. To make this really robust I think we want to key the unranked case on "numSizes == -1" or something more reliable than the pointer being null.

Reply:
Got it, I made that change.